thirdparty/patchmatchnet/models/net.py

```diff
@@ -0,0 +1,365 @@
+"""Implementation of PatchmatchNet
+    reference: https://github.com/FangjinhuaWang/PatchmatchNet
+
+"""
+from typing import Dict, List, Any
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from thirdparty.patchmatchnet.models.module import ConvBnReLU, depth_regression
+from thirdparty.patchmatchnet.models.patchmatch import PatchMatch
+
+
+class FeatureNet(nn.Module):
+    """Feature Extraction Network"""
+
+    def __init__(self):
+        """define and initialize different layers in the network"""
+
+        super(FeatureNet, self).__init__()
+
+        self.conv0 = ConvBnReLU(3, 8, 3, 1, 1)
+        # [B,8,H,W]
+        self.conv1 = ConvBnReLU(8, 8, 3, 1, 1)
+        # [B,16,H/2,W/2]
+        self.conv2 = ConvBnReLU(8, 16, 5, 2, 2)
+        self.conv3 = ConvBnReLU(16, 16, 3, 1, 1)
+        self.conv4 = ConvBnReLU(16, 16, 3, 1, 1)
+        # [B,32,H/4,W/4]
+        self.conv5 = ConvBnReLU(16, 32, 5, 2, 2)
+        self.conv6 = ConvBnReLU(32, 32, 3, 1, 1)
+        self.conv7 = ConvBnReLU(32, 32, 3, 1, 1)
+        # [B,64,H/8,W/8]
+        self.conv8 = ConvBnReLU(32, 64, 5, 2, 2)
+        self.conv9 = ConvBnReLU(64, 64, 3, 1, 1)
+        self.conv10 = ConvBnReLU(64, 64, 3, 1, 1)
+
+        self.output1 = nn.Conv2d(64, 64, 1, bias=False)
+        self.inner1 = nn.Conv2d(32, 64, 1, bias=True)
+        self.inner2 = nn.Conv2d(16, 64, 1, bias=True)
+        self.output2 = nn.Conv2d(64, 32, 1, bias=False)
+        self.output3 = nn.Conv2d(64, 16, 1, bias=False)
+
+    def forward(self, x: torch.Tensor) -> Dict[str, torch.Tensor]:
+        """forward method
+        Args:
+            x: batched of images from one view, in the shape of [B, C, H, W]. Generally, C=3

```

Comment:
Thanks for the updates, Ren. What does "batched of images from one view" mean?

Reply:
If there are N views of images as input, every batch of images from view_i will go through `FeatureNet` and output their features